pkg/kubectl/cmd/get.go

```diff
@@ -168,7 +168,14 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 
 	if len(args) == 0 && cmdutil.IsFilenameEmpty(options.Filenames) {
 		fmt.Fprint(errOut, "You must specify the type of resource to get. ", valid_resources)
-		return cmdutil.UsageError(cmd, "Required resource not specified.")
+
+		fullCmdName := cmd.Parent().CommandPath()
+		if fullCmdName == "" {
```

Comment:
If `fullCmdName == ""` we shouldn't assume that the user can use `kubectl explain`. 

We also need a check to see if `<parent_command> explain` exists, something like:
```go
func isExplainExist(cmd *cobra.Command) bool {
		for _, c := range cmd.Parent().Commands() {
			if c.Name() == "explain" {
				return true
			}
		}
		return false 
}
```

and we only print the suggestion when `len(fullCmdName)>0` (i.e. `fullCmdName != ""`) and the explain command is found:
```go
usageString := "Required resource not specified."
if len(fullCmdName)>0 && isExplainExist(cmd) {
	usageString = fmt.Sprintf("%s\nUse ...", usageString, fullCmdName)
}
```